src/adapter/graphql/src/mutations/dataset_mut/dataset_mut_utils.rs

```diff
@@ -0,0 +1,36 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use kamu_core::DatasetOwnershipService;
+use opendatafabric as odf;
+
+use crate::prelude::*;
+use crate::utils;
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+pub(crate) async fn ensure_account_owns_dataset(
+    ctx: &Context<'_>,
+    dataset_handle: &odf::DatasetHandle,
+) -> Result<()> {
+    let dataset_ownership_service = from_catalog::<dyn DatasetOwnershipService>(ctx).unwrap();
+    let logged_account = utils::get_logged_account(ctx);
+
+    let not_owner = !dataset_ownership_service
+        .is_dataset_owned_by(&dataset_handle.id, &logged_account.account_id)
+        .await?;
```

Comment:
I was hopping we get rid of the `DatasetOwnershipService` completely via ReBAC queries, but I see it's still popular for some reason.

Reply:
I was planning on doing it a little later